packages/@lwc/template-compiler/src/parser/html.ts

```diff
@@ -11,11 +11,26 @@ import { ParserDiagnostics } from '@lwc/errors';
 
 import ParserCtx from './parser';
 import { sourceLocation } from '../shared/ast';
+import { errorCodesToErrorOn, errorCodesToWarnOn } from './parse5Errors';
+
+function getLwcErrorFromParse5Error(code: string) {
+    if (errorCodesToErrorOn.has(code)) {
+        return ParserDiagnostics.INVALID_HTML_SYNTAX;
+    }
+    if (errorCodesToWarnOn.has(code)) {
+        return ParserDiagnostics.INVALID_HTML_SYNTAX_WARNING;
+    }
+    // It should be impossible to reach here; we have a test in parser.spec.ts to ensure
+    // all error codes are accounted for. But just to be safe, make it a warning
+    return ParserDiagnostics.INVALID_HTML_SYNTAX_WARNING;
```

Comment:
This would effectively make this function
```
return errorCodesToErrorOn.has(code)
    ? ParserDiagnostics.INVALID_HTML_SYNTAX
    : ParserDiagnostics.INVALID_HTML_SYNTAX_WARNING
```
or something equivalent right? Do we still need to check both lists?

Reply:
The second list exists because I check it in the unit tests to make sure that all parse5 error codes are accounted for. My idea is that, if we upgrade parse5 and there's a new error, then we need to handle it.

I guess I could do something smarter here, like logging a warning or something. Or maybe I could just have the second list used only in the tests?

Reply:
Another option is to just throw an error, but that makes me a bit nervous.

Reply:
I agree that guardrail is good to have, and just throwing an error as the catch-all makes me nervous too.

I feel that checking the list in both the test and in the code feels unnecessary, but I do see how writing it this way makes the intentions clearer. Would prefer just using the second list in the test, but am okay with it as-is if you prefer that.

Reply:
My 2 cents is that if we do maintain 2 lists, we should have a logging mechanism in place reporting when a code isn't present in `errorCodesToErrorOn` nor `errorCodesToWarnOn`. Otherwise, I would agree with @jye-sf original comment to use a ternary operator.

Reply:
I looked through the template-compiler, and we only use `console.*` in two places:

https://github.com/salesforce/lwc/blob/e27c292d867067197b80ef577f818ced4b361381/packages/%40lwc/template-compiler/src/index.ts#L72-L78

In the absence of a more holistic logging system, I'll just add a `console.warn`.

Reply:
That function (`compileToFunction`) is mainly used for testing, so I'm not sure console.warn will surface properly. I think this would be better as a different diagnostic for "unknown parse5 error code", optionally displayed to the user, with a `level` of `warning` or `log`.

That being said, it's a long weekend and this is for a patch inv, so I'm okay with adding a TODO here so we can get this into master and baking before next week's patch deadline.

Reply:
Yeah I'm not proposing putting the `console.warn` in `compileToFunction`. I'm proposing putting it here:

https://github.com/salesforce/lwc/blob/ece7ef58b353cf28a1959a760307b9b3e4ca1bff/packages/%40lwc/template-compiler/src/parser/html.ts#L20-L27

The user will still get the message, it will just be the same as the other parse5 warnings (i.e. "this will not be supported in a future version of LWC").

Since this line of code should _never_ get hit, unless something has gone wrong with the unit tests or Parse5 has other error codes we don't know about, I feel pretty safe about just adding a `console.warn` and calling it a day.

Reply:
I added a todo linking to #2650

Reply:
Right, agreed. I just meant that since the only use of `console.*` is in a test function, I was unsure whether the log will be easily visible or accessible to us in a production environment.